pkg/model/silence.go

```diff
@@ -45,23 +47,28 @@ func (s *Silence) Validate() error {
 		return fmt.Errorf("silence pattern cannot be empty")
 	}
 
-	if s.TTL == "" {
-		return fmt.Errorf("silence ttl cannot be empty")
-	}
-
 	regex, err := regexp.Compile(s.Pattern)
 	if err != nil {
 		return err
 	}
 	s.RegexPattern = regex
 
-	t, err := common.ParseDuration(s.TTL)
-	if err != nil {
-		return err
+	if s.TTL == "" && !s.CreatedAt.IsZero() && !s.ExpiredAt.IsZero() {
+		durationHours := int(math.RoundToEven(s.ExpiredAt.Sub(s.CreatedAt).Hours()))
```

Comment:
why only `hours` supported?

Reply:
`common.ParseDuration` doesn't support parsing string like `10d2h`. Therefore, we have to convert it to hours then round up.